web/controllers/user_controller.ex

```diff
@@ -37,11 +37,9 @@ defmodule SlackCoder.UserController do
 
   def update(conn, %{"id" => id, "user" => user_params}) do
     user = Repo.get!(User, id)
-    changeset = User.changeset(user, user_params)
 
-    case Repo.update(changeset) do
+    case UserService.save(user, User.changeset(user, user_params)) do
```

Comment:
Take notice the subject of each function. Nested function calls like this should be a code smell... Think of it like this instead:

```
User
|> Repo.get!(id)
|> User.changeset(user_params)
|> UserService.save() # Shouldn't need the user here cause `changeset.data` is the `User`
|> case do
    ...
end
```